File: leetcode/easy/35. Search_Insert_Position/sol.py
from typing import List

def searchInsert(self, nums: List[int], target: int) -> int:
    start = 0
    end = len(nums)-1
    
    while start <= end:
        mid = start + int((end-start)/2)            
        if nums[mid] < target:
            start = mid + 1
        else:
            end = mid - 1
    
    return start

File: leetcode/easy/35. Search-Insert-Position/sol.js
var searchInsert = function (nums, target) {
  let start = 0,
    end = nums.length - 1;
  let middle = 0;
  while (start <= end) {
    middle = parseInt((start + end) / 2);
    if (nums[middle] === target) return middle;

    if (nums[middle] < target) {
      start = middle + 1;
    } else {
      end = middle - 1;
    }
  }
  return nums[middle] > target ? middle : middle + 1;
};
